<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QmlDesigner::Internal::SettingsPage</class>
 <widget class="QWidget" name="QmlDesigner::Internal::SettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>601</width>
    <height>389</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Snapping</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinItemSpacing">
          <property name="maximum">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="snapMarginLabel">
          <property name="text">
           <string>Parent item padding:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinSnapMargin">
          <property name="maximum">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="itemSpacingLabel">
          <property name="text">
           <string>Sibling item spacing:</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Canvas</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="canvasWidthLabel">
          <property name="text">
           <string>Width</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="canvasHeightLabel">
          <property name="text">
           <string>Height</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinCanvasHeight">
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>10000</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinCanvasWidth">
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>10000</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string>Styling</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Controls style:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="styleLineEdit">
          <property name="placeholderText">
           <string>Default style</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetStyle">
          <property name="text">
           <string>Reset Style</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="title">
      <string>QML Emulation Layer</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="useDefaultPuppetRadioButton">
        <property name="toolTip">
         <string>If you select this radio button, Qt Quick Designer always uses the QML emulation layer (QML Puppet) located at the following path.</string>
        </property>
        <property name="text">
         <string>Use fallback QML emulation layer</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Path:</string>
          </property>
          <property name="indent">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Utils::PathChooser" name="fallbackPuppetPathLineEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>Path where Qt Creator can find the QML emulation layer executable (qmlpuppet).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetFallbackPuppetPathButton">
          <property name="toolTip">
           <string>Resets the path to the QML emulation layer that comes with Qt Creator.</string>
          </property>
          <property name="text">
           <string>Reset Path</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QRadioButton" name="useQtRelatedPuppetRadioButton">
        <property name="text">
         <string>Use QML emulation layer that is built by the selected Qt</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Top level build path:</string>
          </property>
          <property name="indent">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Utils::PathChooser" name="puppetBuildPathLineEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetQmlPuppetBuildPathButton">
          <property name="text">
           <string>Reset Path</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>Subcomponents</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="checkBox">
        <property name="text">
         <string>Always save when leaving subcomponent in bread crumb</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Warnings</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QCheckBox" name="designerWarningsCheckBox">
          <property name="toolTip">
           <string>Warns about QML features that are not properly supported by the Qt Quick Designer.</string>
          </property>
          <property name="text">
           <string>Warn about unsupported features in the Qt Quick Designer</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="designerWarningsInEditorCheckBox">
          <property name="toolTip">
           <string>Also warns in the code editor about QML features that are not properly supported by the Qt Quick Designer.</string>
          </property>
          <property name="text">
           <string>Warn about unsupported features of Qt Quick Designer in the code editor</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Debugging</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QCheckBox" name="designerShowDebuggerCheckBox">
          <property name="text">
           <string>Show the debugging view</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="designerEnableDebuggerCheckBox">
          <property name="text">
           <string>Enable the debugging view</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QLineEdit</extends>
   <header location="global">utils/pathchooser.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
